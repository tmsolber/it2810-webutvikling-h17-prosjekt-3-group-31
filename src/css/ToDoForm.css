.modal {
    animation: fade .9s ease-out 0.1s both;
    background-color: rgba(0, 0, 0, 0.49);
    max-width: 500px;
    max-height: 600px;
    margin: 0 auto;
    border-top-left-radius: 3px;
    border-top-right-radius: 3px;
    border-bottom-left-radius: 3px;
    border-bottom-right-radius: 3px;
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}

@keyframes fade{

    0%{
        opacity: 0;
    }
    100%{
        opacity: 1;
    }
}

.backdrop {
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    background-color: rgba(0,0,0,0.7);
    padding: 50px;
}

.form {
    display: block;
    padding: 0.01em 16px;
    padding-top: 10px;
    padding-bottom: 10px;
}

.form .todo-input {
    outline: 0;
    background: #fafafa;
    width: 100%;
    border: 0;
    margin: 0 0 15px;
    padding: 10px;
    border-top-left-radius: 3px;
    border-top-right-radius: 3px;
    border-bottom-left-radius: 3px;
    border-bottom-right-radius: 3px;
    box-sizing: border-box;
    font-size: 16px;
}


.form .todo-input::-webkit-input-placeholder {
    color: black;
    opacity: 0.3;
}

button.close {
    -webkit-appearance: none;
    padding: 0;
    background: transparent;
    border: 0;
    cursor: pointer;
}

.close {
    float: right;
    font-size: 30px;
    font-weight: bold;
    line-height: 1;
    color: white;
    text-shadow: 0 1px 0 #fff;
    filter: alpha(opacity=20);
    opacity: .2;
    margin-right: 5px;
}

.focus-text{
    font-family: sans-serif;
    color: white;
    font-size: 3vw;
    font-weight: bold;
    letter-spacing: -1px;
    line-height: 1;
    text-align: center;
}