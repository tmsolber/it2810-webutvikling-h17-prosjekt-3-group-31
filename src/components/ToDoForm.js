import React, {Component} from 'react';
import '../css/ToDoForm.css';

const ENTER_KEY = 13;

class ToDoForm extends Component {
    constructor() {
        super();
        this.handleSubmit = this.handleSubmit.bind(this);
        this.stopPropagation = this.stopPropagation.bind(this);
    }

    handleSubmit(e){
        if (e.which === ENTER_KEY) {
            e.preventDefault();
            this.props.onUpdate(this.refs.input.value);
            this.refs.input.value = "";
        }
    }

    stopPropagation(e){
        e.stopPropagation();
    }

    render() {
        if (!this.props.show) {
            return null;
        }
        return (
            <div className="backdrop" onClick={this.props.onClose}>
                <div className="modal" onClick={this.stopPropagation}>
                    <button type="button" className="close">
                        <span aria-hidden="true" onClick={this.props.onClose}>&times;</span>
                    </button>
                    <h1 className="focus-text">What is your main focus today?</h1>
                    <div className="form">
                        <input type="text" maxLength={50} ref="input" placeholder="ToDo's..."
                               onFocus={(e) => e.target.placeholder=""} onBlur={(e) => e.target.placeholder="ToDo's..."}
                               onKeyDown={this.handleSubmit} className="todo-input" />
                    </div>
                </div>
            </div>
        );
    }
}

export default ToDoForm;