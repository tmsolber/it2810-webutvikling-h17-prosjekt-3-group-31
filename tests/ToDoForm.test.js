import './globalJSDOM';
import ToDoForm from '../src/components/ToDoForm';
import React from 'react';
import { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

test('to be empty when not shown', () => {
  const component = shallow(<ToDoForm show={false} />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

test('test that modal is shown', () => {
  const component = shallow(<ToDoForm show={true} />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

test('test that onUpdate that sets state gets called', () => {
  const component = mount(<ToDoForm show={true} />);
  const onUpdate = jest.fn();
  component.setProps({ onUpdate });
  const tree = toJson(component);
  const input = component.find('input');
  input.simulate('keydown', { which: 'A' });
  input.simulate('keydown', { which: 13 });

  expect(onUpdate).toBeCalled();
  expect(tree).toMatchSnapshot();
});
