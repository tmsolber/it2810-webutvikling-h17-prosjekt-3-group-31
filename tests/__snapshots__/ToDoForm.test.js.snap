// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test that modal is shown 1`] = `
<div
  className="backdrop"
>
  <div
    className="modal"
    onClick={[Function]}
  >
    <button
      className="close"
      type="button"
    >
      <span
        aria-hidden="true"
      >
        ×
      </span>
    </button>
    <h1
      className="focus-text"
    >
      What is your main focus today?
    </h1>
    <div
      className="form"
    >
      <input
        className="todo-input"
        maxLength={50}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="ToDo's..."
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`test that onUpdate that sets state gets called 1`] = `
<ToDoForm
  onUpdate={[Function]}
  show={true}
>
  <div
    className="backdrop"
    onClick={undefined}
  >
    <div
      className="modal"
      onClick={[Function]}
    >
      <button
        className="close"
        type="button"
      >
        <span
          aria-hidden="true"
          onClick={undefined}
        >
          ×
        </span>
      </button>
      <h1
        className="focus-text"
      >
        What is your main focus today?
      </h1>
      <div
        className="form"
      >
        <input
          className="todo-input"
          maxLength={50}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="ToDo's..."
          type="text"
        />
      </div>
    </div>
  </div>
</ToDoForm>
`;

exports[`to be empty when not shown 1`] = `""`;
